src/cephadm/cephadmlib/container_engines.py

```diff
@@ -45,6 +46,15 @@ def supports_split_cgroups(self) -> bool:
         """Return true if this version of podman supports split cgroups."""
         return self.version >= CGROUPS_SPLIT_PODMAN_VERSION
 
+    @property
+    def unlimited_pids_option(self) -> str:
+        """The option to pass to the container engine for allowing unlimited
+        pids (processes).
+        """
+        if self.version >= PIDS_LIMIT_UNLIMITED_PODMAN_VERSION:
+            return '--pids-limit=-1'
+        return '--pids-limit=0'
```

Comment:
my understanding of the pids limit implementation was that you can set a container to have unlimited pids with docker by setting the limit to 0, and in podman by setting the limit to -1. The old check was
```
if (
        isinstance(ctx.container_engine, Podman)
        and ctx.container_engine.version >= PIDS_LIMIT_UNLIMITED_PODMAN_VERSION
    ):
        container_args.append('--pids-limit=-1')
    else:
        container_args.append('--pids-limit=0')
```
which implies,
"set it to -1 if we're using podman of a version greater than the pids limit unlimited version"
"set it to 0 if we're using docker"
"do not set it at all if we're using podman of a version below the pids limit unlimited version"

After this patch it seems like we'd be setting the limit to 0 for cases with podman versions below the version check where I don't think it supports the flag.

Reply:
I double checked the old code. I agree with your copy-paste. I don't agree with your conclusion.

Maybe a table will help. In the table "is podman" is equivalent to the type check.
The "plupv" value is equivalent to the podman version being >= 
    PIDS_LIMIT_UNLIMITED_PODMAN_VERSION.

```
is podman | plupv  | value
========================
f         | x      |  0
t         | f      |  0
t         | t      | -1
```

note: when is podman is false plupv is a "dont care" because having that value is predicated on is podman being true.

Due to how the else branch is not conditional on any other value any time either "is podman" or "plupv" is false the value will be zero.
There is no case where --pids-limit will not be appended to the list.
    
Let me know what you think.

Reply:
yeah I think you're right. It was supposed to be 0 if it was below the version and -1 otherwise. I went back and verified from the podman release notes that that is the actual correct thing to do as well, instead of just what we had.